src/raster/array.jl

```diff
@@ -1,89 +1,106 @@
+import DiskArrays: eachchunk, GridChunks
+import DiskArrays
 const AllowedXY = Union{Integer,Colon,AbstractRange}
 const AllowedBand = Union{Integer,Colon,AbstractArray}
 
 
+# Define a RasterDataset type
+struct RasterDataset{T,DS} <: AbstractDiskArray{T,3}
+    ds::DS
+    size::Tuple{Int,Int,Int}
+end
+#Forward a few functions
+for f in (:getgeotransform, :nraster, :getband, :getproj,
+  :width, :height, :destroy)
+
+  eval(:($(f)(x::RasterDataset) = $(f)(x.ds)))
+end
+function RasterDataset(ds::AbstractDataset)
+    iszero(nraster(ds)) && throw(ArgumentError("The Dataset does not contain any raster bands"))
+    s = _common_size(ds)
+    RasterDataset{_dataset_type(ds), typeof(ds)}(ds,s)
+end
+function _dataset_type(ds::AbstractDataset)
+  alldatatypes = map(1:nraster(ds)) do i
+    b = getband(ds,i)
+    pixeltype(b)
+  end
+  reduce(promote_type,alldatatypes)
+end
+function _common_size(ds::AbstractDataset)
+  nr = nraster(ds)
+  allsizes = map(1:nr) do i
+    b = getband(ds,i)
+    size(b)
+  end
+  s = unique(allsizes)
+  length(s) == 1 || throw(DimensionMismatch("Can not coerce bands to single dataset, different sizes found"))
+  Int64.((s[1]...,nr))
+end
+getband(ds::RasterDataset,i) = getband(ds.ds,i)
+unsafe_readraster(args...;kwargs...)  = RasterDataset(unsafe_read(args...;kwargs...))
+readraster(s::String;kwargs...) = RasterDataset(read(s;kwargs...))
```

Comment:
Please add docstrings for the following functions:  `RasterDataset(ds::AbstractDataset)`, `readraster(s)`, `getband(ds,i)`, and possibly any other methods meant to be public that I might have missed out.

Reply:
Do you think we need extra docstrings for this? All these functions are already documented for AbstractDataset

Reply:
I think I understand it better now: see https://github.com/yeesian/ArchGDAL.jl/pull/105/files#r411777018. Unfortunately, `RasterDataset` is not a subtype of `AbstractDataset`, so those docstrings will not carry over here.

Reply:
Sorry, I think I did not read your comment well enough and only referred to `getband` which already has a docstring. Otherwise, RasterDataset and and readraster are documented now.